views/api/views.py

```diff
@@ -0,0 +1,34 @@
+from django.http import JsonResponse
+
+from core.user_controller.UserController import UserController
+
+
+def search_user_api(request):
+    """
+    Preconditions:
+    - `request` is a valid HttpRequest object.
+    - `request.GET` contains an optional "query" parameter, which may be an empty string.
+    - `UserController.searchUser` is properly implemented to handle the provided query.
+
+    Postconditions:
+    - If a valid query is provided or left empty, a JSON response containing a list of user data (username and name) is returned.
+    - If an error occurs (e.g., invalid query), a JSON response with an error message is returned with status 400.
+
+    Side-effects:
+    - None.
+
+    Parameters:
+    - request: HttpRequest object, containing the GET data with an optional "query" parameter.
+
+    Returns:
+    - JsonResponse:
+      * On success: A JSON object with a list of users (username and name).
+      * On failure: A JSON object with an "error" key and status code 400.
+    """
+    query = request.GET.get("query", "").strip() if request.GET.get("query", "").strip() else ""
+    try:
+        users = UserController.searchUser(query)
+        user_data = [{"username": user.username, "name": user.name} for user in users]
```

Comment:
Each user object should be a user ref with this data format so you should just be able to use users as user_data